packages/react/src/ReactChildren.js

```diff
@@ -158,14 +159,22 @@ function traverseAllChildrenImpl(
   } else {
     const iteratorFn = getIteratorFn(children);
     if (typeof iteratorFn === 'function') {
-      if (__DEV__) {
+      if (iteratorFn === children.entries) {
```

Comment:
After this diff, this check runs for everyone even outside of DEV. Can we move it inside the condition?